lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -190,6 +253,73 @@ void HWMemSimImplPass::runOnOperation() {
     if (genOp.descriptor() == "FIRRTL_Memory") {
       auto mem = analyzeMemOp(oldModule);
 
+      // If this memory has defined write-after-write behavior, we need to
+      // determine which sets of write ports are driven by the same clock.
+      if (mem.writeUnderWrite == 1) {
```

Comment:
I'd recommend pulling the body of this 'if' out to its own function to make it easier to understand what is going on.

Reply:
Removed entirely due to a rework of this PR.